cluster/mesos/cluster.go

```diff
@@ -169,31 +167,31 @@ func NewCluster(scheduler *scheduler.Scheduler, TLSConfig *tls.Config, master st
 
 // Handle callbacks for the events
 func (c *Cluster) Handle(e *cluster.Event) error {
-	// publish event to the watchQueue for external subscribers
-	c.watchQueue.Publish(e)
-	// call Handle for other eventHandlers
-	c.eventHandlers.Handle(e)
+	// call Handle for all clusterEventHandlers
+	c.clusterEventHandlers.HandleAll(e)
 	return nil
 }
 
 // RegisterEventHandler registers an event handler.
-func (c *Cluster) RegisterEventHandler(h cluster.EventHandler, q *watch.Queue) error {
-	// only set if watchQueue hasn't been set already. This is to avoid issues because
-	// the watchdog code still uses the old event handler.
-	if q != nil && c.watchQueue == nil {
-		c.watchQueue = q
-	}
-	return c.eventHandlers.RegisterEventHandler(h)
+func (c *Cluster) RegisterEventHandler(h cluster.EventHandler) error {
+	return c.clusterEventHandlers.RegisterEventHandler(h)
 }
 
 // UnregisterEventHandler unregisters a previously registered event handler.
 func (c *Cluster) UnregisterEventHandler(h cluster.EventHandler) {
-	c.eventHandlers.UnregisterEventHandler(h)
+	c.clusterEventHandlers.UnregisterEventHandler(h)
+}
+
+// NewAPIEventHandler creates a new API events handler
+func (c *Cluster) NewAPIEventHandler() *cluster.APIEventHandler {
+	return cluster.NewAPIEventHandler()
 }
 
-// CloseWatchQueue closes the watchQueue when the manager shuts down.
-func (c *Cluster) CloseWatchQueue() {
-	c.watchQueue.Close()
+// CloseWatchQueues unregisters all API event handlers (the ones with
+// watch queues) and closes the respective queues. This should be
+// called when the manager shuts down
+func (c *Cluster) CloseWatchQueues() {
```

Comment:
Oh, this is a mistake I think. It was supposed to call the `CloseWatchQueues` function of `ClusterEventHandlers`.